product/roundhouse.tasks/Roundhouse.cs

```diff
@@ -174,6 +187,11 @@ public string RecoveryMode
 
         public void run_the_task()
         {
+            if (this.loggingHelper.HasLoggedErrors)
+            {
+                return;
+            }

```

Comment:
Also not always MSBuild coming through here, so probably better not to explicitly use this call here.